prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -297,19 +292,27 @@ class ReportFunction : Logging {
     }
 
     private fun processAsync(
-        report: Report,
+        parsedReport: Report,
         workflowEngine: WorkflowEngine,
         options: Options,
         defaults: Map<String, String>,
         routeTo: List<String>,
         actionHistory: ActionHistory
     ) {
-        // add 'Process' queue event to the actionHistory
+
+        val report = parsedReport.copy()
+
+        if (report.bodyFormat != Report.Format.INTERNAL) {
+            error("Processing a non internal report async.")
+        }
+
         val processEvent = ProcessEvent(Event.EventAction.PROCESS, report.id, options, defaults, routeTo)
-        actionHistory.trackEvent(processEvent)
 
+        val blobInfo = workflowEngine.blob.uploadBody(report, "processing", processEvent.eventAction)
```

Comment:
Need to remove this magic string and decide where these blobs should live

Reply:
See BlobAccess.uploadBody (the second override) - the magic strings all live in one place and are calculated based off of Event.EventAction. Probably should just add a .PROCESS option into the blobName when statement.